src/main/java/org/generationcp/middleware/api/brapi/VariableServiceBrapiImpl.java

```diff
@@ -0,0 +1,260 @@
+		if (!CollectionUtils.isEmpty(variable.getStudyDbIds())) {
+			for (final String studyDbId : variable.getStudyDbIds()) {
+				this.addObservationVariableToStudy(Integer.valueOf(studyDbId), Integer.valueOf(variable.getObservationVariableDbId()),
+					variable.getObservationVariableName());
+			}
+		}
+		return variable;
+	}
+
+	private void updateVariable(final VariableDTO variable, final boolean isVariableUsedInStudy) {
+		final Integer variableId = Integer.valueOf(variable.getObservationVariableDbId());
+		final CVTerm variableTerm = this.daoFactory.getCvTermDao().getById(variableId);
+		if (!isVariableUsedInStudy) {
+			// Updating of name, property, scale and method relationship is only applicable
+			// if the variable is not used in study.
+			variableTerm.setName(variable.getObservationVariableName());
+			this.updatePropertyMethodScaleRelation(variable);
+		}
+		this.daoFactory.getCvTermDao().update(variableTerm);
+	}
+
+	private void updatePropertyMethodScaleRelation(final VariableDTO variable) {
+		// load scale, method and property relationships
+		final int variableId = Integer.parseInt(variable.getObservationVariableDbId());
+		final List<CVTermRelationship> relationships = this.daoFactory.getCvTermRelationshipDao().getBySubject(variableId);
+		final Optional<CVTermRelationship> methodRelation =
+			relationships.stream().filter(r -> Objects.equals(r.getTypeId(), TermRelationshipId.HAS_METHOD.getId())).findAny();
+		final Optional<CVTermRelationship> propertyRelation =
+			relationships.stream().filter(r -> Objects.equals(r.getTypeId(), TermRelationshipId.HAS_PROPERTY.getId())).findAny();
+		final Optional<CVTermRelationship> scaleRelation =
+			relationships.stream().filter(r -> Objects.equals(r.getTypeId(), TermRelationshipId.HAS_SCALE.getId())).findAny();
+
+		if (propertyRelation.isPresent()) {
+			final CVTermRelationship cvTermRelationship = propertyRelation.get();
+			cvTermRelationship.setObjectId(Integer.valueOf(variable.getTrait().getTraitDbId()));
+			this.daoFactory.getCvTermRelationshipDao().update(cvTermRelationship);
+		}
+		if (methodRelation.isPresent()) {
+			final CVTermRelationship cvTermRelationship = methodRelation.get();
+			cvTermRelationship.setObjectId(Integer.valueOf(variable.getMethod().getMethodDbId()));
+			this.daoFactory.getCvTermRelationshipDao().update(cvTermRelationship);
+		}
+		if (scaleRelation.isPresent()) {
+			final CVTermRelationship cvTermRelationship = scaleRelation.get();
+
+			final List<CVTermRelationship> dataTypeRelationShips = this.daoFactory.getCvTermRelationshipDao()
+				.getBySubjectIdAndTypeId(cvTermRelationship.getObjectId(), TermId.HAS_TYPE.getId());
+
+			// Update Datatype
+			if (!CollectionUtils.isEmpty(dataTypeRelationShips) && StringUtils.isNotEmpty(variable.getScale().getDataType())) {
+				final CVTermRelationship dataTypeRelationShip = dataTypeRelationShips.get(0);
+				dataTypeRelationShip.setSubjectId(Integer.valueOf(variable.getScale().getScaleDbId()));
```

Comment:
I think we are missing the line here to actually update the data type of the scale (ie. dataTypeRelationship.setObjectId)? We should also perform checking that data type is valid before updating